main.sh

```diff
@@ -1,5 +1,36 @@
 #!/usr/bin/env bash
-
+set -xe
+
+function check_dependencies {
+  for x in git sed awk; do
+    which ${x} 1>&2 >/dev/null
+	if [[ $? -ne 0 ]]; then
+	  printf "dependency check failed! ${x} not found in PATH\n"
+	  exit 2
+	fi
+  done
+  return 0
+}
+# this implements future options "post-clone"
+function post_setup {
+  local config_source=~/.start-project/config #  ${XDG_CONFIG_HOME}/start-project/config
+  if [[ -f ${config_source} && -z $broken_enabled ]]; then
+    #such a hack begin_ and end_ are a poor man's delimiter! ugh this isn't good, but should work for now
+	# not working as well as I hoped, I am keeping this here to share the idea
+    local commands=$(awk '/begin_post_setup_steps/{ getline cmd; if (cmd ~ /end_post_setup_steps/){ exit;}; print "\047"cmd"\047", cmd }' $config_source)
```

Comment:
This is pretty cool.  Is there a way to keep `config_source` in the project template?  That way they could be customized per project.  We'd have trash them while we're burning the Git evidence, too.

Reply:
We should add an example to the README.

Reply:
hmm yeah I like where your going with that, it really should be configurable per project, but I'm not thinking of anything right now. Including it in the README or as a --switch/parameter could be another option. Environment variable too, but they tend to stick around.

Reply:
```
start-project/
  config - global stuff like where templates are stored, ssh identities, whatever
  templates/
    bar/
    foo/
      start-project-install.sh - optional project-specific scripts to run
```

1. We need a non-stupid name.
2. We need to exclude it from the new project.  Maybe.